source/plugin/blocks/accessing.rst

```diff
@@ -34,56 +34,104 @@ referenced block is any kind of banner by checking the blocks type.
+    import org.spongepowered.api.data.manipulator.immutable.ImmutableWetData;
 
-    public boolean isWet(Location block) {
-        if (block.getBlock().getType() != BlockTypes.SPONGE) {
+    public void isWet(Location blockLoc) {
+        BlockState sponge = blockLoc.getBlock();
+        if (!sponge.getType().equals(BlockTypes.SPONGE)) {
             return false;
         }
-        BlockState sponge = block.getBlock();
-        Optional<WetData> wetness = sponge.getManipulator(WetData.class);
+        Optional<ImmutableWetData> wetness = sponge.get(ImmutableWetData.class);
         return wetness.isPresent();
     }
 
-First, we need to know which ``DataManipulator`` subinterface we need. Those that are applicable to blocks are found in
-the ``org.spongepowered.api.data.manipulator`` and ``org.spongepowered.api.data.manipulator.block`` packages. Then we
-can just pass that class to the ``getManipulator()`` method and get an ``Optional`` in return which will be ``empty()``
-if the ``BlockState`` does not contain data of that type. Since ``WetData`` represents a boolean value, its presence
-equates to ``true``. Its absence (if ``Optional.empty()`` was returned) either signifies ``false``.
+More information on mutable and immutable ``DataManipulator`` s can be found in the :doc:`data documentation <../data/datamanipulators>`.
+
+Block Properties
+~~~~~~~~~~~~~~~~
+Blocks can contain certain properties. A property is a pre-set value that defines the game logic of that particular block. For example, blocks can contain pre-determined blast-resistance values that can be used to determine what you're working with, without actually checking the type of block it could be one by one. For example, if we wanted to get the blast resistance of a block and checking if it is greater than or equal to one, it would be done like so:
+
+.. code-block:: java
+
+    import org.spongepowered.api.data.property.block.BlastResistanceProperty;
+
+    public boolean blastResistance(Location<World> blockLoc) {
+        Optional<BlastResistanceProperty> optional =
+            blockLoc.getProperty(BlastResistanceProperty.class);
+        
+        if(optional.isPresent()) {
+            BlastResistanceProperty resistance = optional.get();
+            if(resistance.getValue() >= 1) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+A list of possible block properties can be found in the package ``org.spongepowered.api.data.property.block``.
+
+Block Traits
+~~~~~~~~~~~~
+A block trait is a certain value on the current state of a block. A block may or may not contain block traits depending on the type of block. For example, a bed has a ``BooleanTrait`` called ``BED_OCCUPIED``. As a boolean can only have two values, true and false, the ``BED_OCCUPIED`` trait can only be true or false. Checking this value is simple, just call the ``getTraitValue()`` method on a ``BlockState``. An example of this with a bed is shown below:
```

Comment:
Mention that BlockTraits are only really meant as a fallback, and where possible DataManipulators are the preferred method of dealing with this data.

Traits really are only a fallback for modded compatibility (I think, I might be wrong @gabizou )